packages/repository/src/relations/has-many/has-many-through.repository.ts

```diff
@@ -191,15 +192,29 @@ export class DefaultHasManyThroughRepository<
       constrainFilter(undefined, sourceConstraint),
       options?.throughOptions,
     );
-    const targetFkValues = this.getTargetKeys(throughInstances);
-    // delete through instances that have the targets that are going to be deleted
-    const throughFkConstraint = this.getThroughConstraintFromTarget(
-      targetFkValues,
-    );
-    await throughRepository.deleteAll(
-      constrainWhereOr({}, [sourceConstraint, throughFkConstraint]),
-    );
-
+    if (where) {
+      // only delete related through models
+      const targets = await targetRepository.find({where: where});
```

Comment:
Since we already have a list of `troughInstances`, is there a way how to limit this query to find only target models that are matching the user-provided `where` and are related to the source model(s)?

I am concerned about performance implication of the current solution, because it may fetch too many records. Consider the case Doctor-Appointment-Patient case, where we want to delete all appointment+patient records of the given doctor where the patient has `{died: true}`. Let's say our database is large (e.g. ~100k doctors). For each doctor, there will be only few patients that have passed away (e.g. 1-10). Ideally, we should fetch only those 1-10 patients, not ~100k-1000k passed-away patients of all doctors.

I am fine to leave this performance optimization out of scope of the initial implementation, as long as we have some documentation for this known performance issue (a code comment is good enough).

A nitpick on coding style:
```suggestion
      const targets = await targetRepository.find({where});
```

Reply:
Agreed 👍. I couldn't find a way to fix it without accessing to the database, but we can improve it by limiting how many records it fetches.

I think it needs more helpers to build such a where filter. I left a TODO comment.